java/impl/aws/src/main/java/co/worklytics/psoxy/S3Handler.java

```diff
@@ -0,0 +1,106 @@
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import lombok.SneakyThrows;
+import lombok.extern.java.Log;
+import org.apache.commons.io.input.BOMInputStream;
+
+import javax.inject.Inject;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+@Log
+public class S3Handler implements com.amazonaws.services.lambda.runtime.RequestHandler<S3Event, String> {
+
+    @Inject
+    StorageHandler storageHandler;
+
+    @Inject
+    ConfigService configService;
+
+    @Inject
+    AmazonS3 s3Client;
+
+    @SneakyThrows
+    @Override
+    public String handleRequest(S3Event s3Event, Context context) {
+
+        DaggerAwsContainer.create().injectS3Handler(this);
+
+        boolean isBOMEncoded = Boolean.parseBoolean(configService.getConfigPropertyAsOptional(AWSConfigProperty.BOM_ENCODED).orElse("false"));
+        String destinationBucket = configService.getConfigPropertyAsOptional(AWSConfigProperty.OUTPUT_BUCKET)
+                .orElseThrow(() -> new IllegalStateException("Output bucket not found as environment variable!"));
+
+        S3EventNotification.S3EventNotificationRecord record = s3Event.getRecords().get(0);
+
+        String importBucket = record.getS3().getBucket().getName();
+        String sourceKey = record.getS3().getObject().getUrlDecodedKey();
+
+        log.info(String.format("Received a request for processing %s from bucket %s. BOM flag is marked as %s", sourceKey, importBucket, isBOMEncoded));
+
+        S3Object s3Object = s3Client.getObject(new GetObjectRequest(importBucket, sourceKey));
+        StorageEventResponse storageEventResponse;
+
+        try(InputStream objectData = s3Object.getObjectContent()) {
+            InputStreamReader reader;
+
+            BOMInputStream is = null;
+            if (isBOMEncoded) {
+                is = new BOMInputStream(objectData);
+                reader = new InputStreamReader(is, StandardCharsets.UTF_8.name());
+            } else {
+                reader = new InputStreamReader(objectData);
+            }
+
+            StorageEventRequest request = StorageEventRequest.builder()
+                    .sourceBucketName(importBucket)
+                    .sourceObjectPath(sourceKey)
+                    .destinationBucketName(destinationBucket)
+                    .readerStream(reader)
+                    .build();
+
+           storageEventResponse = storageHandler.handle(request);
+
+           if (isBOMEncoded) {
+               is.close();
+           }
+           reader.close();
+        }
```

Comment:
You can always call with `BOMInputStream`, if it contains the mark it will exclude it.
So is safe to call it in either case.
You can remove `isBOMEncoded` and everywhere where it is call I think (like AWSConfigProperty.BOM_ENCODED)

```suggestion
        try(InputStream objectData = s3Object.getObjectContent(); 
            InputStreamReader reader = new InputStreamReader(new BOMInputStream(objectData), StandardCharsets.UTF_8.name())) {

            StorageEventRequest request = StorageEventRequest.builder()
                    .sourceBucketName(importBucket)
                    .sourceObjectPath(sourceKey)
                    .destinationBucketName(destinationBucket)
                    .readerStream(reader)
                    .build();

           storageEventResponse = storageHandler.handle(request);
        }
```